doom-modeline.el

```diff
@@ -262,6 +262,14 @@ So it can be restored when `doom-modeline-mode' is disabled.")
               (with-current-buffer bname
                 (doom-modeline-set-main-modeline))))
 
+        ;; Many themes do not set the header-line font correctly, this is just
+        ;; extra precaution
+        (when doom-modeline-header-line
+            (custom-set-faces
+             '(header-line ((t (:foreground nil :background nil :inherit mode-line))))
```

Comment:
It's not a good idea to customize faces here. It's better to define some extra faces to do that.